lib/reek/smells/smell_warning.rb

```diff
@@ -8,11 +8,13 @@ module Smells
     class SmellWarning
       include Comparable
       extend Forwardable
-      attr_reader :context, :lines, :message, :parameters, :smell_detector
-      def_delegators :smell_detector, :smell_category, :smell_type, :source
+      attr_reader :context, :lines, :message, :parameters, :smell_detector, :source
+      attr_reader :source
+      def_delegators :smell_detector, :smell_category, :smell_type
 
       def initialize(smell_detector, options = {})
         @smell_detector = smell_detector
+        @source         = options.fetch(:source)
```

Comment:
This technically is an API change, but we do not describe smell warning creation in the API docs.